src/strategies/flex_window.py

```diff
@@ -84,170 +86,321 @@ def step(self, event_list=[]):
-                # vehicle events ignored (use vehicle info such as estimated_time_of_departure)
+                    # vehicle events ignored (use vehicle info such as estimated_time_of_departure)
 
             # get (predicted) external load
-            if timestep_idx == 0:
-                # use actual external load
-                ext_load = gc.get_current_load()
-                # add battery power (sign switch, as ext_load is subtracted)
-                ext_load -= avail_bat_power
+            # if timestep_idx == 0:
+            #     # use actual external load
+            #     ext_load = gc.get_current_load()
+            # else:
+                # get external load
+            if self.EXT_LOAD == "perfect_foresight":
+                ext_load = self.events.get_external_load_perfect_forseight(
+                    start_time=cur_time, interval=self.interval
+                ) - sum(cur_feed_in.values())
             else:
-                ext_load = gc.get_avg_ext_load(cur_time, self.interval) - sum(cur_feed_in.values())
-            timesteps.append({
-                "power": cur_max_power - ext_load,
-                "ext_load": ext_load,
-                "window": cur_window,
-            })
-
-        if util.get_cost(1, gc.cost) <= self.PRICE_THRESHOLD:
-            # charge max
-            commands = {}
-            for vehicle in self.world_state.vehicles.values():
-                cs_id = vehicle.connected_charging_station
-                if cs_id is None:
-                    continue
-                cs = self.world_state.charging_stations[cs_id]
-                p = gc.cur_max_power - gc.get_current_load()
-                p = util.clamp_power(p, vehicle, cs)
-                avg_power = vehicle.battery.load(self.interval, p)['avg_power']
-                commands[cs_id] = gc.add_load(cs_id, avg_power)
-                cs.current_power += avg_power
-        elif self.LOAD_STRAT == 'balanced':
-            commands = self.distribute_balanced(timesteps)
+                ext_load = gc.get_avg_ext_load(cur_time, self.interval) - sum(
+                    cur_feed_in.values()
+                )
+            # save infos for each timestep
+            vehicles = sorted(
+                [
+                    v
+                    for v in self.world_state.vehicles.values()
+                    if v.connected_charging_station is not None
+                ],
+                key=self.sort_key,
+            )
+
+            timesteps.append(
+                {
+                    "timestep_idx": timestep_idx,
+                    "power": cur_max_power - ext_load,
+                    "ext_load": ext_load,
+                    "window": cur_window,
+                    "v_load": 0,
+                    "total_load": ext_load
+                }
+            )
+            for i, row in enumerate(timesteps):
+
+                for v in vehicles:
+                    timesteps[i].update({v.connected_charging_station: 0})
```

Comment:
remove blank line. add comment what this is doing. Reading it as "save status of all charging stations for each timestep, defaults to 0". What happens if a charging station has a name like one of the other timesteps keys (e.g. window or power), since they are user-controlled? Better to have own sub-dictionary: timestep -> charging_stations -> all cs keys

Reply:
left it out as it's not needed anyways